src/main/java/com/mineboundteam/minebound/inventory/AlloyFurnaceMenu.java

```diff
@@ -0,0 +1,225 @@
+
+            if (itemstack1.isEmpty()) {
+                slot.set(ItemStack.EMPTY);
+            } else {
+                slot.setChanged();
+            }
+
+            if (itemstack1.getCount() == itemstack.getCount()) {
+                return ItemStack.EMPTY;
+            }
+
+            slot.onTake(pPlayer, itemstack1);
+        }
+
+        return itemstack;
+    }
+
+
+    // Copied from AbstractFurnaceMenu because slots are set up in the constructor
+    // and java is a BAD language and minecraft is a POORLY written game. Fuck Mojang and fuck Oracle
+    protected final Container container;
+    protected final ContainerData data;
+    protected final Level level;
+    protected final RecipeType<? extends AbstractCookingRecipe> recipeType;
+    protected final RecipeBookType recipeBookType;
+ 
+    public void fillCraftSlotsStackedContents(StackedContents pItemHelper) {
+       if (this.container instanceof StackedContentsCompatible) {
+          ((StackedContentsCompatible)this.container).fillStackedContents(pItemHelper);
+       }
+ 
+    }
+ 
+    public void clearCraftingContent() {
+       this.getSlot(0).set(ItemStack.EMPTY);
+       this.getSlot(2).set(ItemStack.EMPTY);
+    }
+ 
+    public boolean recipeMatches(Recipe<? super Container> pRecipe) {
+       return pRecipe.matches(this.container, this.level);
+    }
+ 
+    public int getResultSlotIndex() {
+       return 2;
+    }
+ 
+    public int getGridWidth() {
+       return 1;
+    }
+ 
+    public int getGridHeight() {
+       return 1;
+    }
+ 
+    public int getSize() {
+       return 3;
+    }
+ 
+    /**
+     * Determines whether supplied player can use this container
+     */
+    public boolean stillValid(Player pPlayer) {
+       return this.container.stillValid(pPlayer);
+    }
+    
+    protected boolean canSmelt(ItemStack pStack) {
+       return this.level.getRecipeManager().getRecipes().stream().anyMatch((r) -> r.getIngredients().stream().anyMatch((i) -> i.test(pStack)));
+    }
+ 
+    protected boolean isFuel(ItemStack pStack) {
+       return net.minecraftforge.common.ForgeHooks.getBurnTime(pStack, this.recipeType) > 0;
+    }
+ 
+    public int getBurnProgress() {
+       int i = this.data.get(AbstractFurnaceBlockEntity.DATA_COOKING_PROGRESS);
+       int j = this.data.get(AbstractFurnaceBlockEntity.DATA_COOKING_TOTAL_TIME);
+       return j != 0 && i != 0 ? i * 24 / j : 0;
+    }
+ 
+    public int getLitProgress() {
+       int i = this.data.get(1);
+       if (i == 0) {
+          i = 200;
+       }
+ 
+       return this.data.get(0) * 13 / i;
+    }
+ 
+    public boolean isLit() {
+       return this.data.get(0) > 0;
+    }
+ 
+    public RecipeBookType getRecipeBookType() {
+       return this.recipeBookType;
+    }
+ 
+    public boolean shouldMoveToInventory(int pSlotIndex) {
+       return pSlotIndex != 1;
+    }
+
+    // This game makes me want to strangle someone
+    protected class FuckingFuelSlot extends FurnaceFuelSlot{
```

Comment:
Should probably rename this once the rage has passed.

Reply:
_make  me_